pkg/minishift/openshift/oc.go

```diff
@@ -42,3 +46,24 @@ func AddSudoersRoleForUser(user string) error {
 	}
 	return nil
 }
+
+// Add Current Profile Context
+func AddContextForProfile(profile string, ip string) error {
+	cmdName := instanceState.Config.OcPath
+	ip = strings.Replace(ip, ".", "-", -1)
+	cmdArgs := []string{"config", "set-context", profile,
+		fmt.Sprintf("--cluster=%s:%d", ip, constants.APIServerPort),
+		fmt.Sprintf("--user=developer/%s:%d", ip, constants.APIServerPort),
+		"--namespace=myproject"}
```

Comment:
For what it's worth, the following works for me and does not rely on a namespace:

    $ oc config set-context hardy --cluster=192-168-99-106:8443 --user=developer/192-168-99-106:8443

Also, looking at `oc config view`, it seems the --cluster is referring to the cluster name, which if you look at the config is:

     - cluster:
        certificate-authority-data: REDACTED
        server: https://192.168.99.106:8443
      name: 192-168-99-106:8443

So the name is uses actually hyphens not an actual IPs with dots. This is also what one should refer to in the various commands (unless of course one also set the cluster name as mentioned in the previous comment). 

My vote would be to create a 'minishift' cluster alias as well and then use this name to create the context. Also the actual minishift context does imo not need a namespace.